### PR TITLE
intel-aero-image: Add some developer tools

### PR DESCRIPTION
**Pull request by zehortigoza**
Adding valgrind, strace, gdb and cgdb.

### CODE REVIEW

File: recipes-core/images/intel-aero-image.bb [hunk 1]
```diff
@@ -38,6 +38,9 @@ IMAGE_INSTALL += "libavahi-client libavahi-glib"
 # Build tools
 IMAGE_INSTALL += "packagegroup-core-buildessential"
 
+# Development/Debug tools
+IMAGE_INSTALL += "valgrind strace gdb cgdb"
```
Comment (lucasdemarchi) [inline recipes-core/images/intel-aero-image.bb]: strace and gdb are already there.

valgrind, ok

cgdb would pull dependencies we currently don't have.... not sure it's worth. You can always start gdb as server and use graphical tools in your host (e.g. https://gcc.gnu.org/onlinedocs/gcc-4.8.3/gnat_ugn_unw/Remote-Debugging-using-gdbserver.html).

Comment (lucasdemarchi) [inline recipes-core/images/intel-aero-image.bb]: I had a typo in my `find` command... we actually have all the dependencies for cgdb already. I still think it's not worth though.

**Comment by zehortigoza** [comment]
Ok, patch changed to just add valgrind.

**Comment by whbruce** [comment]
The image is steadily growing with developer tool requests such as this. I recommend putting all such tools in a packagegroup and either creating a new developer image or extend existing image with [IMAGE_EXTRA_FEATURES](http://www.yoctoproject.org/docs/2.1/dev-manual/dev-manual.html#usingpoky-extend-customimage-imagefeatures).

**Comment by mindThomas** [comment]
What about catkin tools? Will they be a part of the image anytime soon as well?
Or any description on how to cross-compile ROS nodes would be very welcome.

**Comment by lucasdemarchi** [comment]
> The image is steadily growing with developer tool requests such as this. I recommend putting all such tools in a packagegroup and either creating a new developer image or extend existing image with IMAGE_EXTRA_FEATURES.

I understand that. The difference from most other layers here is that we are "the final layer". The tools we put on the image are there to cover the common *developer* use case, which is the focus of the board. Most of the people aren't building another image based on Yocto with their additional tools, but rather using whatever we are giving them.

Once we have packages working, we could shrink the size of the image and just publish the packages so people could easily install them. Until that happens, I don't see the image shrinking... so the tools we ship in the image are on a as-needed basis.

That said, we could create a core-image with only the bare minimal (it's also good to allow us to check size). However intel-aero-image must be the image we publish as official.

**Comment by lucasdemarchi** [comment]
> What about catkin tools? Will they be a part of the image anytime soon as well?

We are very busy with other parts of the distros... adding or not adding will mostly depend if the size grows a little bit or explode in side. If someone sends a PR for that, then it's an easier decision.

> Or any description on how to cross-compile ROS nodes would be very welcome.

That depends on a published esdk... should be done soon, but I can't promise when.

**Pull request merged**